common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -41,34 +44,33 @@
    * @param buffer the byte buffer to be unmapped
    */
   public static void cleanDirectBuffer(ByteBuffer buffer) {
-    if (buffer == null) {
-      return;
-    }
-    if (buffer.isDirect()) {
-      try {
-        if (sCleanerMethod == null) {
-          sCleanerMethod = buffer.getClass().getMethod("cleaner");
-          sCleanerMethod.setAccessible(true);
-        }
-        final Object cleaner = sCleanerMethod.invoke(buffer);
-        if (cleaner != null) {
-          if (sCleanMethod == null) {
-            sCleanMethod = cleaner.getClass().getMethod("clean");
-          }
-          sCleanMethod.invoke(cleaner);
-        }
-      } catch (Exception e) {
-        LOG.warn("Fail to unmap direct buffer due to ", e);
-        buffer = null;
+    Preconditions.checkNotNull(buffer);
+    Preconditions.checkArgument(buffer.isDirect(), "buffer isn't a DirectByteBuffer");
```

Comment:
Both locations that call this method are passing a direct byte buffer currently, so it is not breaking code currently.
Its name seems to suggest this is a direct buffer expected? 
I am open to change for this. Just to see what you think more reasonable.